wavefront_lambda/__init__.py

```diff
@@ -63,7 +75,19 @@ def wavefront_wrapper(*args, **kwargs):
         if not server:
             raise ValueError("Environment variable WAVEFRONT_URL is not set.")
         auth_token = os.environ.get('WAVEFRONT_API_TOKEN')
-        if not auth_token:
+        # WAVEFRONT_PROXY != 0 use WavefrontProxyReporter. default WavefrontDirectReporter
+        proxy = False
+        try:
+            int(os.getenv('WAVEFRONT_PROXY', "0"))
+        except ValueError:
+            pass
+        finally:
+            proxy = bool(proxy)
+        # WAVEFRONT_PORT PORT default 2878
+        port = os.environ.get("WAVEFRONT_PORT", 2878)
+        # support WAVEFRONT_PREFIX from environment
+        prefix = os.environ.get("WAVEFRONT_PREFIX", "")
```

Comment:
Not sure of a use case for setting a custom wavefront_prefix. Can be removed.
All the metrics reported by this lambda wrapper are considered standard lambda metrics with a well defined prefix "aws.lambda.wf" . We also have an OOTB lambda dashboard in wavefront built on this naming convention for all standard lambda metrics.

Reply:
the wavefront_prefix is optional. if it's not provided from the environment, old behaver (prefix=''') is still used. The reason to give user option to apply prefix is for the use case if the user is using wavefront with the application (directly or using other agents to collect/push metrics) and needs to correlate metrics between "AWS" and Application. 

Giving an example, we are currently using Telegraf to collect application metrics. Since some of our applications are running in AWS, we also take metrics from CloudWatch. We have the need to correlate data between AWS CloudWatch and application, sharing similar prefix can make later Wavefront (UI) query/aggregation much easier.

Reply:
So just to clarify,  do you need the standard lambda metrics reported by the lambda wrapper with a custom prefix as well ? 
The standard lambda metrics are at :  https://github.com/wavefrontHQ/wavefront-lambda-python/blob/master/README.md#standard-lambda-metrics-reported-by-wavefront-lambda-wrapper

Reply:
@ruivapps 
**I think we are talking about two different prefix here.

What you describe is the “METRICS_PREFIX” in call_lambda_with_standard_metrics function.
You have delta.delta_counter(reg, METRIC_PREFIX + string) for MetricsRegistry.

I did not change anything on this part. The part I did update is when create WaveFrontReporter instance.**
[Anil]
I understand your change. The reporter at https://github.com/wavefrontHQ/wavefront-lambda-python/blob/master/wavefront_lambda/__init__.py#L98 is responsible to report all standard lambda metrics along with custom metrics. The prefix that you specify here applies to all metrics.

Ex:
1. For prefix="" : 
 standard lambda metrics = "aws.lambda.wf.invocations.count" and
 custom metrics = "<whatever_name_you_specify>"

2. For prefix='custom1' : 
 standard lambda metrics  = "custom1.aws.lambda.wf.invocations.count" and
 custom metrics = "<custom1.whatever_name_you_specify>"

If you do not want to modify the standard lambda metrics, you should not include any prefix here. Instead, you can just have a function in your lambda code that adds the prefix to the metric name when you create a new metric (Counter / Gauge) etc.

Having said that I am ok with this change. Would be good if you add few more comments to clarify the above as per my other review comments. Just in case you do decide to use the WAVEFRONT_PREFIX then you can clone our OOTB lambda dashboards and modify them to use the new WAVEFRONT_PREFIX to retrieve metrics. 

**In WavefrontProxyReporter, the default prefix is “proxy.” and for WavefrontDirectReporter the default prefix is “direct.” (https://github.com/wavefrontHQ/wavefront-pyformance/blob/master/wavefront_pyformance/wavefront_reporter.py line 106 and 133)**
[Anil] Currently prefix="" , so reporter will not take the default values that you are referring to above.